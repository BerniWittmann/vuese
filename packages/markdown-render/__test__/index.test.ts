import Render, { RenderResult, MarkdownResult } from '@vuese/markdown-render'
import { ParserResult } from '@vuese/parser'

test('Proper rendering of the table header', () => {
  const res: ParserResult = {
    name: 'MyComponent',
    componentDesc: {
      default: ['This is a description of the component'],
      group: ['My Group']
    },
    props: [
      {
        name: 'someProp',
        type: ['String'],
        typeDesc: ['`TOP` / `BOTTOM`'],
        required: true,
        defaultDesc: ['`TOP`'],
        describe: ['Represents the direction of the arrow']
      }
    ],
    events: [
      {
        name: 'click',
        isSync: false,
        syncProp: '',
        describe: ['Triggered when clicked'],
        argumentsDesc: ['a boolean value']
      }
    ],
    slots: [
      {
        name: 'header',
        describe: 'Table header',
        backerDesc: '`<th>{{title}}</th>`',
        bindings: {},
        scoped: false
      }
    ],
    methods: [
      {
        name: 'clear',
        describe: ['Clear form'],
        argumentsDesc: ['a boolean value']
      }
    ]
  }
  const render = new Render(res)
  const renderRes: RenderResult = render.render()
  const markdownRes = render.renderMarkdown() as MarkdownResult
  expect(renderRes).toMatchSnapshot()
  expect(markdownRes).toMatchSnapshot()
})
