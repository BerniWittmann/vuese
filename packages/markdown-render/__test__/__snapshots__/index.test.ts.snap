// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Components (without slots,methods,props,events) can be forced to be rendered Does render an empty component by with @vuese in the description 1`] = `
Object {
  "componentName": "MyComponent",
  "content": "# MyComponent

This is a description of the component

",
}
`;

exports[`Empty Components (without slots,methods,props,events) can be forced to be rendered Does render an empty component by with @vuese in the description 2`] = `Object {}`;

exports[`Proper rendering of the table header 1`] = `
Object {
  "events": "|Event Name|Description|Parameters|
|---|---|---|
|click|Triggered when clicked|a boolean value|
",
  "methods": "|Method|Description|Parameters|
|---|---|---|
|clear|Clear form|a boolean value|
",
  "props": "|Name|Description|Type|Required|Default|
|---|---|---|---|---|
|someProp|Represents the direction of the arrow|\`TOP\` / \`BOTTOM\`|\`true\`|\`TOP\`|
",
  "slots": "|Name|Description|Default Slot Content|
|---|---|---|
|header|Table header|\`<th>{{title}}</th>\`|
",
}
`;

exports[`Proper rendering of the table header 2`] = `
Object {
  "componentName": "MyComponent",
  "content": "# MyComponent

This is a description of the component

## Props

<!-- @vuese:MyComponent:props:start -->
|Name|Description|Type|Required|Default|
|---|---|---|---|---|
|someProp|Represents the direction of the arrow|\`TOP\` / \`BOTTOM\`|\`true\`|\`TOP\`|

<!-- @vuese:MyComponent:props:end -->


## Events

<!-- @vuese:MyComponent:events:start -->
|Event Name|Description|Parameters|
|---|---|---|
|click|Triggered when clicked|a boolean value|

<!-- @vuese:MyComponent:events:end -->


## Slots

<!-- @vuese:MyComponent:slots:start -->
|Name|Description|Default Slot Content|
|---|---|---|
|header|Table header|\`<th>{{title}}</th>\`|

<!-- @vuese:MyComponent:slots:end -->


## Methods

<!-- @vuese:MyComponent:methods:start -->
|Method|Description|Parameters|
|---|---|---|
|clear|Clear form|a boolean value|

<!-- @vuese:MyComponent:methods:end -->


",
}
`;
